go/arrow/flight/flightsql/client.go

```diff
@@ -396,6 +408,11 @@ func (c *Client) Prepare(ctx context.Context, query string, opts ...grpc.CallOpt
 		return
 	}
 	defer stream.CloseSend()
+	defer func() {
+		// XXX: assuming server will not return a result and then an error
+		// (or else we need to also try to clean up stmt)
+		err = readUntilEOF(stream)
+	}()
```

Comment:
would it make more sense to put this into `parsePreparedStatementResponse` and get it for free in all the spots that just delegate to it?